Libs/Requirement/RequirementFactory.cs

```diff
@@ -90,54 +90,85 @@ public Requirement GetRequirement(string name, string requirement)
                     {  "Drinking", ()=> playerReader.Buffs.Drinking },
                     {  "Mana Regeneration", ()=> playerReader.Buffs.ManaRegeneration },
-                    {  "Fortitude", ()=> playerReader.Buffs.Fortitude },
-                    {  "InnerFire", ()=> playerReader.Buffs.InnerFire },
-                    {  "Divine Spirit", ()=> playerReader.Buffs.DivineSpirit },
-                    {  "Renew", ()=> playerReader.Buffs.Renew },
-                    {  "Shield", ()=> playerReader.Buffs.Shield },
+                    //-------------------------
+
+
+                    // Druid
                     {  "Mark of the Wild", ()=> playerReader.Buffs.MarkOfTheWild },
                     {  "Thorns", ()=> playerReader.Buffs.Thorns },
                     {  "TigersFury", ()=> playerReader.Buffs.TigersFury },
+                    //-------------------------
+                    {  "Demoralizing Roar", ()=> playerReader.Debuffs.Roar },
+                    {  "Faerie Fire", ()=> playerReader.Debuffs.FaerieFire },
+                    {  "Rip", ()=> playerReader.Debuffs.Rip },
+
+                    // Hunter
+
+                    //-------------------------
+
+                    // Mage
                     {  "Frost Armor", ()=> playerReader.Buffs.FrostArmor },
                     {  "Arcane Intellect", ()=> playerReader.Buffs.ArcaneIntellect },
                     {  "Ice Barrier", ()=>playerReader.Buffs.IceBarrier },
                     {  "Ward", ()=>playerReader.Buffs.Ward },
                     {  "Fire Power", ()=>playerReader.Buffs.FirePower },
+                    //-------------------------
+                    {  "Frostbite", ()=> playerReader.Debuffs.Rip },
+
+                    //  Paladin
+                    {  "Seal", ()=> playerReader.Buffs.Seal },
+                    {  "Aura", ()=>playerReader.Buffs.Aura },
+                    {  "Devotion Aura", ()=>playerReader.Buffs.Aura },
+                    {  "Blessing", ()=> playerReader.Buffs.Blessing },
+                    {  "Blessing of Might", ()=> playerReader.Buffs.Blessing },
+                    //-------------------------
+
+                    // Priest
+                    {  "Fortitude", ()=> playerReader.Buffs.Fortitude },
+                    {  "InnerFire", ()=> playerReader.Buffs.InnerFire },
+                    {  "Divine Spirit", ()=> playerReader.Buffs.DivineSpirit },
+                    {  "Renew", ()=> playerReader.Buffs.Renew },
+                    {  "Shield", ()=> playerReader.Buffs.Shield },
+                    //-------------------------
+                    {  "Shadow Word: Pain", ()=> playerReader.Debuffs.ShadowWordPain },
+
+                    // Rogue
                     {  "Slice And Dice", ()=> playerReader.Buffs.SliceAndDice },
-                    {  "Battle Shout", ()=> playerReader.Buffs.BattleShout },
-                    {  "Demon Skin", ()=> playerReader.Buffs.DemonSkin },
```

Comment:
You can crunch the `Demon Skin` and `Demon Armor` buffs into a single bit.
However with this change, all of the warlock profiles below level 20 are not gonna work properly.
Gonna throw Missing requirements `Demon Skin`.

What you can do here to do not break the profiles is to involve some craft here.

`{ "Demon Skin", ()=> playerReader.Buffs.DemonArmor },`

diff
```
diff --git a/Libs/Requirement/RequirementFactory.cs b/Libs/Requirement/RequirementFactory.cs
index 767c2a6..a6acee9 100644
--- a/Libs/Requirement/RequirementFactory.cs
+++ b/Libs/Requirement/RequirementFactory.cs
@@ -147,6 +147,7 @@ namespace Libs

                     // Warlock
                     {  "Demon Armor", ()=> playerReader.Buffs.DemonArmor },
+                    {  "Demon Skin", ()=> playerReader.Buffs.DemonArmor },
                     {  "Soul Link", ()=> playerReader.Buffs.SoulLink },
                     {  "Soulstone Resurrection", ()=> playerReader.Buffs.SoulstoneResurrection },
                     {  "Shadow Trance", ()=> playerReader.Buffs.ShadowTrance },
```